oci/layout/oci_src_test.go

```diff
@@ -0,0 +1,133 @@
+package layout
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"testing"
+
+	"github.com/containers/image/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const RemoteLayerContent = "This is the remote layer content"
+
+func TestMain(m *testing.M) {
+	httpServer, serverStopSignal, err := startRemoteLayerServer()
+	if err != nil {
+		println("Error starting test TLS server", err.Error())
+		os.Exit(1)
+	}
+
+	code := m.Run()
+
+	stopServer(httpServer, serverStopSignal)
+	os.Exit(code)
+}
+
+func TestGetBlobForRemoteLayers(t *testing.T) {
+	imageSource := createImageSource(t, &types.SystemContext{})
+	layerInfo := types.BlobInfo{
```

Comment:
(Non-blocking: Strictly speaking, the `Digest` field is mandatory and `Size` should be `-1` if unknown. OTOH I realize that computing the digests would be fairly pointless pain.)